gCoffee/app/src/main/java/com/mborowiec/gcoffee/Order.java

```diff
@@ -108,21 +109,36 @@ private void makeOrder(String coffee) {
     }
 
     /**
-     * Checks whether the menu contains the ordered item. If it does, asks to confirm order.
+     * Checks whether the menu contains the ordered item.
+     * @param item  item ordered by user
+     * @return boolean value
+     */
+    public boolean checkMenu(String item, String[] coffeeList) {
+        if (item != null) item = item.toLowerCase();
+        for (int i = 0; i < coffeeList.length; i++) {
+            coffeeList[i] = coffeeList[i].toLowerCase();
+        }
+
+        return Arrays.asList(coffeeList).contains(item);
+    }
+
+    /**
+     * Handles the incoming deeplink with ordered item. Checks whether the item is on the menu.
+     * If it is, asks to confirm order, if it is not, notifies the user about it.
+     * If the input is null, it  starts the Main activity instead of proceeding to order.
      * @param data incoming deeplink
      */
     private void handleDeepLinks(Uri data) {
-            String coffee;
-            coffee = data.getQueryParameter("inMenuName");
-            String[] coffeeList = this.getResources().getStringArray(R.array.coffee_list);
-            boolean contains = Arrays.asList(coffeeList).contains(coffee);
+        String coffee = data.getQueryParameter("inMenuName");
+        if (coffee == null) startActivity(new Intent(getApplicationContext(), MainActivity.class));
```

Comment:
It's a coding style question... but these very long one liners seem to be hard to read/error prone... I would recommend using {} and a new line

Reply:
Thanks, I will change it